File: 2026/February-10-2026/max_average_subarray.js

nums = [1,12,-5,-6,50,3]
k = 4

const max_average_sub = (nums, k) => { 
  let curr = 0; 
  
  for(let i = 0; i < k; i++){ 
    curr += nums[i]
  }
  
  max_array = curr

  for(i = k; i < nums.length; i++) {
    curr += nums[i] - nums[i - k]
    max_array = Math.max(max_array, curr)
  }
  return max_array / k
}

console.log(max_average_sub(nums, k))

File: 2026/February-10-2026/max_average_subarray.py
# Input: nums = [1,12,-5,-6,50,3], k = 4
# Output: 12.75000
# Explanation: Maximum average is (12 - 5 - 6 + 50) / 4 = 51 / 4 = 12.75

from typing import List

nums = [1,12,-5,-6,50,3]
k = 4

def max_average_subarray(nums: List[int], k: int) -> float: 
  curr = 0
  
  for right in range(0, k): 
    curr += nums[right]

  max_subarray = curr

  for right in range(k, len(nums)): 
    curr += nums[right] - nums[right - k]

    max_subarray = max(max_subarray, curr)

  return max_subarray / k


print(max_average_subarray(nums, k))